### PR TITLE
updated to exclude tags in sar metrics

### PR DESCRIPTION
**Pull request by gnegri**

### CODE REVIEW

File: common/reportMetrics.py [hunk 1]
```diff
@@ -444,9 +446,15 @@ def process_replay(file_path):
 
         else:  # metric file replay processing
             # handle different metric agents
-            if parameters['agentType'] == 'sar':
-                for command in ['sar', 'sar -n DEV', 'sar -p -d']:
-                    replay_sar(file_path, command)
+            if 'sar' in parameters['agentType']:
```
Comment (yqian10) [inline common/reportMetrics.py]: now that you have already determine whether 'sar' in parameters['agentType'], why you put or condition below?

File: common/reportMetrics.py [hunk 2]
```diff
@@ -518,7 +526,7 @@ def replay_sar(metric_file_path, command):
         # parse opening lines
         line = metric_file.readline()
         header = line.split()
-        instance = header[2][1:-1]
+        host = header[2][1:-1]
```
Comment (yqian10) [inline common/reportMetrics.py]: please throw exception/ pre check when deal with substring

File: common/reportMetrics.py [hunk 3]
```diff
@@ -537,6 +545,8 @@ def replay_sar(metric_file_path, command):
         row_count = 0
         min_timestamp_epoch = 0
         max_timestamp_epoch = -1
+        exclude_tags = agent_config_vars['exclude_tags']
+        logger.debug(exclude_tags)
```
Comment (yqian10) [inline common/reportMetrics.py]: Please add more detailed message in the debug mode like "exclude tags: " + str(exclude_tags)

File: common/reportMetrics.py [hunk 4]
```diff
@@ -559,9 +569,21 @@ def replay_sar(metric_file_path, command):
...
+                if len(exclude_tags) != 0:
+                    for tag in exclude_tags:
+                        if device == tag or (tag.endswith('*') and device.startswith(tag.split('*')[0])):
+                            exclude_flag = True
+                            break
+                instance += '_' + device
+            if exclude_flag:
+                line = metric_file.readline()
+                continue
+            logger.debug(instance)
```
Comment (yqian10) [inline common/reportMetrics.py]: same here

File: common/reportMetrics.py (hunk 1 above)
Comment (gnegri) [reply common/reportMetrics.py]: -t sar will do each of sar-cpu, sar-mem, sar-network, and sar-storage

File: common/reportMetrics.py [hunk 5]
```diff
@@ -526,6 +526,9 @@ def replay_sar(metric_file_path, command):
         # parse opening lines
         line = metric_file.readline()
         header = line.split()
+        if len(header) < 4:
+            logger.error('Error in sar file header')
+            sys.exit(1)
         host = header[2][1:-1]
```
Comment (yqian10) [inline common/reportMetrics.py]: header[2][1:-1] is to trim the first and last character.
ex. 3453465423 -> 45346542.
is this what you want?

Comment (gnegri) [reply common/reportMetrics.py]: yes, the string is "(host)"

**Pull request merged**